src/geometry_base/strategy_base.cc

```diff
@@ -38,19 +38,11 @@ namespace geometry_base {
 bool IsNewRoadPositionResultCloser(const maliput::api::RoadPositionResult& new_road_position_result,
                                    const maliput::api::RoadPositionResult& road_position_result) {
   const double delta = new_road_position_result.distance - road_position_result.distance;
-  const bool different_segment = road_position_result.road_position.lane->segment()->id() !=
-                                 new_road_position_result.road_position.lane->segment()->id();
-
-  // When lanes belong to the same segment is expected that the distance value is almost equal so we can't use the
-  // distance as main condition. When lanes don't belong to the same segment we can use the distance as main
-  // condition.
-  if (different_segment) {
```

Comment:
Is it ok to change this behavior?

Reply:
Just asking because this was something implemented a lot of time ago. I'd double-check if the maliput_malidrive's tests and maliput_integration_tests are still passing.

Reply:
I've checked it locally and everything works. The short circuit of checking for the same segment was before we introduced the ToSegment and ToLane position conversions. For this case, it is important to have ToLanePosition rather than ToSegment to avoid mappings outside the scope of the Route / Phase. That said, do you want me to create branches pointing to this patch?

Reply:
Ah I see, thanks for the explanation. No need to address it separately